satpy/composites/__init__.py

```diff
@@ -191,8 +212,9 @@ def check_geolocation(self, data_arrays):
         areas = [ds.attrs.get('area') for ds in data_arrays]
         if all(a is None for a in areas):
             return
-        elif any(a is None for a in areas):
-            raise ValueError("Missing 'area' attribute")
+        for ds in data_arrays:
+            if "area" not in ds.attrs:
+                raise AttributeError(f"Dataset {ds.name!s} missing 'area' attribute")
```

Comment:
I'm not so sure this being an AttributeError versus ValueError. It is in `.attrs` which are technically "attributes", but not the typical `obj.attribute` ones.

Also, this should be `ds.attrs['name']` in the format string. `ds.name` is likely going to be the dask array's task name which isn't going to be super useful.

Reply:
About the AttributeError, I guess I don't care enough to ask for you to change it, but not sure it is perfect.

Reply:
According to the `xarray.DataArray` documentation, the name is an attribute of the `DataArray`:

http://xarray.pydata.org/en/stable/generated/xarray.DataArray.name.html

```
In [85]: da = xarray.DataArray([1, 2], name="abc")

In [86]: da.name
Out[86]: 'abc'

In [87]: da.attrs["name"]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-87-1743ae92d127> in <module>
----> 1 da.attrs["name"]

KeyError: 'name'
```

All `xarray.DataArray` objects have names (possibly `None`), but they do not always have a `name` entry in the `attrs` dict.

You have a good point that NetCDF attributes are not Python attributes, but in that case they're rather keys in the `.attrs` dictionary, so then it should be a KeyError.  But then again, Python attributes are also just keys in a dictionary somewhere under the hood in Python.

Reply:
I haven't tested with xarray dataarrays wrapping dask arrays though.  Do those overwrite the name of the data array?

Reply:
I've changed it to `ds.attrs.get('name', ds.name)`.

Reply:
Yes, dask array names are usually used inplace of the DataArray option. The DataArray created from a Dataset created from xr.open_dataset will typically have the name of the NetCDF variable. However, `.attrs['name']` should **always** exist in Satpy and is the name that a Satpy user would understand. The dask or DataArray `.name` will not be useful in most cases.